### PR TITLE
pybind/rados: add rados_service_*()

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

File: src/pybind/rados/rados.pyx [hunk 1]
```diff
@@ -1493,6 +1496,38 @@ Rados object in state %s." % self.state)
...
+    def service_daemon_register(self, service, daemon, metadata):
+        """
+        :param str service: service name (e.g. "rgw")
+        :param str daemon: daemon name (e.g. "gwfoo")
+        :param dict metadata: static metadata about the register daemon
+               (e.g., the version of Ceph, the kernel version.)
+        """
+        service = cstr(service, 'service')
+        daemon = cstr(daemon, 'daemon')
+        metadata_dict = '\0'.join(chain.from_iterable(metadata.items()))
```
Comment (jdurgin) [inline src/pybind/rados/rados.pyx]: do these dict inputs need a trailing '\0' added for the c call?

Comment (tchaikov) [reply src/pybind/rados/rados.pyx]: i inspected the concat'ed string using gdb, it does have two trailing `\0`s.
```
(gdb) p/c (char[24])*metadata_dict
$12 = {118 'v', 101 'e', 114 'r', 115 's', 105 'i', 111 'o', 110 'n', 0 '\000', 51 '3', 46 '.', 49 '1', 52 '4', 0 '\000',
  109 'm', 101 'e', 109 'm', 111 'o', 114 'r', 121 'y', 0 '\000', 52 '4', 50 '2', 0 '\000', 0 '\000'}
```

but you are right. we can not rely on this.

**Comment by tchaikov** [comment]
@jdurgin fixed and repushed.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-12-05_05:34:25-rados-wip-kefu-testing-2017-12-05-1155-distro-basic-mira/

**Pull request merged**